src/commands/Checkin.ts

```diff
@@ -191,7 +191,11 @@ export default class Checkin extends Command {
     const qrCodes: MessageAttachment[] = [];
 
     // For each event we are given...
-    await events.forEach(async (event) => {
+    // Note: We use Promise.all since each async function call in map sends a Promise.
+    // We do this because we need to await qrCodeDataUrl's return value before pushing to qrCodes.
+    // forEach doesn't allow async callbacks and for ... of doesn't allow us to run in parallel,
+    // so this solution works most effectively + efficiently and is still linting-friendly.
+    await Promise.all(events.map(async (event) => {
```

Comment:
This is a nice approach to handling concurrent async actions mapping through a list! 

I'm not 100% sure if this is what you intended so I just want to clarify one small difference. What happens if any of the promises reject?

`Promise.all` resolves if **all** of the promises resolve successfully which is what we want however it short-circuits and stops early as soon as any promise inside throws an error and rejects, which might cancel the remaining ones.

`Promise.any` resolves if **any** of the promises resolve and short-circuits as soon as we get one that resolves.

`Promise.allSettled` might be the more ideal balance for what we want. It'll take all of the promises and run all of them even if a couple fail.

Found an [article](https://medium.com/@volodymyrfrolov/beware-of-short-circuiting-promise-combinators-in-javascript-bbb5b7a9e70f) on that explains it a bit more